src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/HttpSource.cs

```diff
@@ -317,6 +317,11 @@ public Task<JObject> GetJObjectAsync(Uri uri, ILogger log, CancellationToken tok
                             await UpdateHttpClient(promptCredentials);
                             continue;
                         }
+
+                        // null means cancelled by user
+                        // block subsequent attempts to annoy user with prompts
+                        _authRetries = HttpHandlerResourceV3Provider.MaxAuthRetries;
```

Comment:
During an install this would end up coming up constantly. For different UI actions it may make sense to retry for each. Do you have a specific scenario where a user might be unable to provide credentials and have to continue on but later have the correct password?